gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -134,6 +134,8 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
       if (jobStatus != null) {
         try(Timer.Context context = getMetricContext().timer(GET_AND_SET_JOB_STATUS).time()) {
           addJobStatusToStateStore(jobStatus, this.stateStore);
+        } catch (IOException ioe) {
+          throw new RuntimeException("Failed to add job status to state store", ioe);
```

Comment:
Add a comment here explaining why we are propagating the exception.

Reply:
Also why do we two catch IOException blocks? Isn't the outer catch block for the exception thrown by addJobStatusToStateStore?

Reply:
The outer one is catching failures to parse job status from the kafka message, which we are treating differently from failure to store the job status in the state store.

Reply:
parseJobStatus method can be modified to swallow the IOException. It seems to be doing that for most part anyways. The IOException being thrown by readSchemaVersioningInformation can be caught inside the method.